fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/event/election/CastVoteFragmentTest.java

```diff
@@ -200,4 +231,30 @@ public void castVoteSendsACastVoteMessage() {
     verify(messageSenderHelper.getMockedSender())
         .publish(any(), eq(ELECTION.getChannel()), any(CastVote.class));
   }
+
+  @Test
+  public void castEncryptedVoteTest() {
+    launchSecretElection();
+
+    onView(withText(ELECTION_BALLOT_TEXT11)).perform(click());
+    castVotePager().perform(swipeLeft());
+    onView(withText(ELECTION_BALLOT_TEXT22)).perform(click());
+    castVoteButton().perform(click());
+    // Wait for the operations performed above to complete
+    InstrumentationRegistry.getInstrumentation().waitForIdleSync();
+
+    encryptedVoteText().check(matches(isDisplayed()));
+  }
+
+  private void launchSecretElection() {
+    fragmentRule
+        .getScenario()
+        .onActivity(
+            activity ->
+                activity
+                    .getSupportFragmentManager()
+                    .beginTransaction()
+                    .replace(containerId(), CastVoteFragment.newInstance(SECRET_ELECTION_ID))
+                    .commit());
+  }
```

Comment:
This feels dirty. Though I don't have a better idea at the moment. Maybe you have one @GabrielFleischer ?

Reply:
That's ok, but I think the best would be to move this test to a new class where the scenario rule would create a CastVote with a Secret Election id.